### PR TITLE
Check that class has only one constructor

### LINKED ISSUE #3: Check that class has only one constructor

### PR DESCRIPTION
**Pull request by artem-zinnatullin**
Closes #3.

@nikitin-da PTAL

### CODE REVIEW

File: private-constructor-checker/src/main/java/com/pushtorefresh/private_constructor_checker/PrivateConstructorChecker.java [hunk 1]
```diff
@@ -61,15 +61,16 @@ public void check() {
...
-        } catch (NoSuchMethodException e) {
-            throw new RuntimeException("Can not get default declared constructor for class = "
-                    + clazz,
-                    e
-            );
+        if (constructors.length > 1) {
+            throw new AssertionError("Class has more than one constructor");
+        }
+
+        final Constructor<?> constructor = constructors[0];
```
Comment (nikitin-da) [inline private-constructor-checker/src/main/java/com/pushtorefresh/private_constructor_checker/PrivateConstructorChecker.java:65]: Can we receive OutOfBounds here?

Comment (nikitin-da) [inline private-constructor-checker/src/main/java/com/pushtorefresh/private_constructor_checker/PrivateConstructorChecker.java:65]: No, we can't)

File: private-constructor-checker/src/test/java/com/pushtorefresh/private_constructor_checker/PrivateConstructionCheckerTest.java [hunk 2]
```diff
@@ -38,13 +37,11 @@ public void shouldThrowExceptionIfClassDoesNotHaveDeclaredConstructors() {
                     .check();
```
Comment (nikitin-da) [inline private-constructor-checker/src/test/java/com/pushtorefresh/private_constructor_checker/PrivateConstructionCheckerTest.java:20]: Maybe it should be better to rename this test to `shouldThrowExceptionIfClassHasNonDefaultConstructor`?

Comment (artem-zinnatullin) [inline private-constructor-checker/src/test/java/com/pushtorefresh/private_constructor_checker/PrivateConstructionCheckerTest.java:20]: Yep, I'll do :)

**Comment by artem-zinnatullin** [comment]
@nikitin-da  I've fixed all comments, PTAL :)

**Comment by nikitin-da** [comment]
Cool!

**Pull request merged**